csrc/multidevice/utils.cpp

```diff
@@ -53,9 +29,33 @@ std::vector<IterDomain*> getShardedIterDomains(TensorView* tv) {
       [](auto id) { return id->isDeviceDim(); });
   return sharded_ids;
 }
-
 } // namespace
 
+bool isSharded(TensorView* tv) {
+  bool is_sharded = false;
+  auto rids = TensorDomain::noReductions(tv->getMaybeRFactorDomain());
+  auto ids = TensorDomain::noReductions(tv->getLeafDomain());
```

Comment:
I think I mentioned this before, but in nvFuser parallelization matters for reduction domains. I don't think it's a good idea to keep the semantic discrepancy as it would be more difficult to correct it later.

Reply:
Makes sense, I will keep this in mind.

I think what we want from isSharded in general is to check there is a device parallel IterDomain on at least any non-reduction IterDomain, because [i0, r(DIDx(i1)] is not a sharded tensorview.

For our error checking we also have:
1. At most 1 non-reduction IterDomain can be parallelized on DIDx and 1 reduction IterDomain can be parallelized on DIDx. 
2. (Temporary) No splits/merges on any DIDx parallelized IterDomain.